lib/output.js

```diff
@@ -117,15 +117,37 @@ function OutputStream(options) {
         function quote_double() {
             return '"' + str.replace(/\x22/g, '\\"') + '"';
         }
+        function quote_template() {
+            if (!options.ascii_only) {
+                str = str.replace(/\\(r|n|u2028|u2029)/g, function(s, c) {
+                    switch(c) {
+                        case "n": return "\n";
+                        case "r": return "\r";
+                        case "u2028": return "\u2028";
+                        case "u2029": return "\u2029";
+                    }
+                    return "\\" + c;
+                });
+            }
+            return '`' + str.replace(/`/g, '\\`') + '`';
+        }
         if (options.ascii_only) str = to_ascii(str);
         switch (options.quote_style) {
           case 1:
+            if (quote == "`" && sq > 0)
+                return quote_template();
             return quote_single();
           case 2:
+            if (quote == "`" && dq > 0)
+                return quote_template();
             return quote_double();
           case 3:
+            if (quote == "`")
+                return quote_template();
             return quote == "'" ? quote_single() : quote_double();
           default:
+            if (quote == "`" && dq > 0 && sq > 0)
```

Comment:
```
tag`This should also be single quoted`
```

Meh, it's just representation (if it can evaluate it will in the compressor). And it's hard to calculate what's best. In the end we should rely on statistics based on real code, which isn't there yet as es6/es7 isn't full swing yet.

Also I'm relying that the output is allowed to be es6.